docs/gates.md

```diff
@@ -16,18 +16,14 @@ print(CNOT(q0, q1))
 # CNOT((0, 0), (0, 1))
 ```
 
-### Gate Features
-
-The raw ``Gate`` class itself simply describes that a ``Gate``
-can be applied to qubits to produce an ``Operation``. We then
-use marker classes for ``Gates`` indicated what additional
-features a ``Gate`` has.  
-
-For example, one feature is ``ReversibleEffect``.
-A ``Gate`` that inherits this class is required to implement the method ``inverse`` which returns the inverse gate.
-Algorithms that operate on gates can use ``isinstance(gate, ReversibleEffect)`` to determine whether gates implements ``inverse`` method, and then use it.
-(Note that, even if the gate is not reversible, the algorithm may have been given an ``Extension`` with a cast from the gate to ``ReversibleEffect``.
-See the [extensions documentation](extensions.md) for more information.)
+### Magic Methods
+
+A raw ``Gate`` class can be applied to qubits to produce an ``Operation``, but that's about it.
```

Comment:
I would remove ", but that's about it".  It seems very informal.

Reply:
Fair point. That's my style of writing, even for technical things. Maybe especially for technical things. I would worry more about the documentation having "many voices" before I worried about it being informal.

This all has to be more extensively rewritten (e.g. the very next section is back to talking about extensions) so I won't worry about it too much here.